packages/data/src/components/with-select/test/index.js

```diff
@@ -671,13 +668,12 @@ describe( 'withSelect', () => {
 				</RegistryProvider>
 			);
 		} );
-		// 4 times:
+		// 3 times:
 		// - 1 on initial render
 		// - 1 on effect before subscription set.
-		// - 1 on re-render
 		// - 1 on effect before new subscription set (because registry has changed)
-		expect( mapSelectToProps ).toHaveBeenCalledTimes( 4 );
-		expect( OriginalComponent ).toHaveBeenCalledTimes( 2 );
+		expect( mapSelectToProps ).toHaveBeenCalledTimes( 3 );
+		expect( OriginalComponent ).toHaveBeenCalledTimes( 3 );
```

Comment:
Verifying the changes from updating the tests, the other test updates make sense to me, but this one doesn't. I'm not sure why `OriginalComponent` renders an extra time as a result of these changes. And it seems to happen any time a dependencies array is provided to `useSelect`, regardless of whether it's a fixed value or not (i.e. still renders an extra time when providing `[]`). I'm not sure if this is a specific characteristic of `useCallback` with dependencies (possibly related to https://github.com/facebook/react/issues/14099?) and whether it really only has impact under a changing registry context that this test case is verifying against. It wouldn't be quite as worrying if it is only for the changing registry, since that doesn't happen quite so often in real-world usage.

Reply:
The first render is for the registry changing, and the second one is for the return value of `mapSelectToProps` changing, right?

It seems to check out. We could also add a test that changes the state without changing the registry and compare.

Reply:
The first render is for the registry changing, and the second one is for the return value of `mapSelectToProps` changing, right?

It seems to check out. We could also add a test that changes the state without changing the registry and compare.

Reply:
> The first render is for the registry changing, and the second one is for the return value of `mapSelectToProps` changing, right?

Yes, I think that's what the original tests were trying to verify. But now there's a third unexplained render.

Reply:
It was only called once before.

Reply:
The mock count carries on from the previous render.

Reply:
I'm referring to the change of:

_Before:_

```js
expect( OriginalComponent ).toHaveBeenCalledTimes( 2 );
```

_After:_

```js
expect( OriginalComponent ).toHaveBeenCalledTimes( 3 );
```

Reply:
Mmmm, have you tried seeing if `ownProps` is somehow changed/invalidated? That could trigger an extra render.